pkg/webhook/server/server.go

```diff
@@ -326,10 +327,23 @@ func (s *Server) mutate(obj runtime.Object) (runtime.Object, error) {
 }
 
 func (s *Server) convert(obj runtime.Object) (runtime.Object, error) {
-	review := obj.(*apiextensionsv1beta1.ConversionReview)
+	outputVersion := apiextensionsv1.SchemeGroupVersion
+	review, isV1 := obj.(*apiextensionsv1.ConversionReview)
+	if !isV1 {
+		outputVersion = apiextensionsv1beta1.SchemeGroupVersion
+		reviewv1beta1 := obj.(*admissionv1beta1.AdmissionReview)
+		convertedReview, err := defaultScheme.ConvertToVersion(reviewv1beta1, admissionv1.SchemeGroupVersion)
+		if err != nil {
+			return nil, err
+		}
+		review = convertedReview.(*apiextensionsv1.ConversionReview)
+	}
+
 	resp := s.ConversionWebhook.Convert(review.Request)
 	review.Response = resp
-	return review, nil
+
+	versionedOutput, err := defaultScheme.ConvertToVersion(review, outputVersion)
+	return versionedOutput, err
```

Comment:
Why did this function change so much? Is this the function that converts  cmapi resources between versions? 
The original never called `defaultScheme.ConvertToVersion`, so how did it used to work?

Also it looks like the function now supports both v1beta1 and v1 apiextensions, but should  we use a type switch to catch those two expected versions and default with a returned error to catch anything else.

Reply:
This does not convert them, it passes the conversion API request to the code that actually handles the conversion. In the past the Kubernetes API could only send a request of type v1beta1. Now it can send both, v1beta1 in legacy installs and v1 in "mainline" installs. Thus the need for us to convert the requests and responses when replying ot k8s.

Reply:
Type switches I think won't work here as the input is a runtime.Object, but I added a better error

Reply:
Got it. Thanks for explaining.